resources/metadata/whitesource.yaml

```diff
@@ -0,0 +1,523 @@
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: https://github.com/whitesource/unified-agent-distribution/releases/latest/download/${config.agentFileName}
+    - name: configFilePath
+      type: string
+      description: Explicit path to the WhiteSource Unified Agent configuration file.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: ./wss-unified-agent.config
+    - name: vulnerabilityReportFileName
+      type: string
+      description: Name of the file the vulnerability report is written to.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: piper_whitesource_vulnerability_report
+    - name: parallelLimit
+      type: string
+      description: 'Limit of parallel jobs being run at once in case of `scanType:
+        ''mta''` based scenarios, defaults to `15`.'
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: 15
+    - name: reporting
+      type: bool
+      description: Whether assessment is being done at all, defaults to `true`.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: true
+    - name: whitesource/serviceUrl
+      type: string
+      description: URL to the WhiteSource server API used for communication, defaults to `https://saas.whitesourcesoftware.com/api`.
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: https://saas.whitesourcesoftware.com/api
+    - name: buildDescriptorExcludeList
+      type: string
+      description: List of build descriptors and therefore modules to exclude from the scan and assessment activities.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: []
+    - name: whitesource/orgToken
+      type: string
+      description: WhiteSource token identifying your organization.
+      mandatory: true
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: licensingVulnerabilities
+      type: bool
+      description: Whether license compliance is considered and reported as part of the assessment.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: true
+    - name: agentFileName
+      type: string
+      description: Locally used name for the Unified Agent jar file after download.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: wss-unified-agent.jar
+    - name: emailAddressesOfInitialProductAdmins
+      type: string
+      description: The list of email addresses to assign as product admins for newly created WhiteSource products.
+      mandatory: false
+      scope:
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: []
+    - name: whitesource/productVersion
```

Comment:
I think `whitesource/` prefixed parameters should be grouped here in the yaml.

Reply:
The slashes were actually producing compile-errors with the generated code so I removed them.